library/azure_ad_users.py

```diff
@@ -0,0 +1,310 @@
+        if not self.login_url:
+            self.login_url = "https://login.windows.net/{}/oauth2/token?api-version=1.0".format(self.tenant_domain)
+
+        # Geting azure cred from ENV if not defined
+        if not self.client_id:
+            if 'azure_client_id' in os.environ:
+                self.client_id = os.environ['azure_client_id']
+            elif 'AZURE_CLIENT_ID' in os.environ:
+                self.client_id = os.environ['AZURE_CLIENT_ID']
+            elif 'client_id' in os.environ:
+                self.client_id = os.environ['client_id']
+            elif 'CLIENT_ID' in os.environ:
+                self.client_id = os.environ['CLIENT_ID']
+            else:
+                # in case client_id came in as empty string
+                self.module.fail_json(msg="Client ID is not defined in module arguments or environment.")
+
+        if not self.client_secret:
+            if 'azure_client_secret' in os.environ:
+                self.client_secret = os.environ['azure_client_secret']
+            elif 'AZURE_CLIENT_SECRET' in os.environ:
+                self.client_secret = os.environ['AZURE_CLIENT_SECRET']
+            elif 'client_secret' in os.environ:
+                self.client_secret = os.environ['client_secret']
+            elif 'CLIENT_SECRET' in os.environ:
+                self.client_secret = os.environ['CLIENT_SECRET']
+            else:
+                # in case secret_key came in as empty string
+                self.module.fail_json(msg="Client Secret is not defined in module arguments or environment.")
+        self.headers = None
+        self.data = None
+        self.azure_version = "api-version=1.6"
+
+    # TODO: might not be needed
+    def convert(self, data):
+        if isinstance(data, basestring):
+            return str(data)
+        elif isinstance(data, collections.Mapping):
+            return dict(map(self.convert, data.iteritems()))
+        elif isinstance(data, collections.Iterable):
+            return type(data)(map(self.convert, data))
+        else:
+            return data
+
+    def login(self):
+        headers = { 'User-Agent': 'ansible-azure-0.0.1', 'Connection': 'keep-alive', 'Content-Type': 'application/x-www-form-urlencoded' }
+        payload = { 'grant_type': 'client_credentials', 'client_id': self.client_id, 'client_secret': self.client_secret }
+        payload = urllib.urlencode(payload)
+
+        try:
+            r = open_url(self.login_url, method="post", headers=headers ,data=payload)
+        except urllib2.HTTPError, err:
+            response_code = err.getcode()
+            response_msg = err.read()
+            response_json = json.loads(response_msg)
+            self.module.fail_json(msg="Failed to login error code = '{}' and message = {}".format(response_code, response_msg))
+
+        response_msg = r.read()
+        # TODO: Should try and catch if failed to seriolize to json
```

Comment:
I would make try catch for JSON serialization as its possible to get html from azure and that will break the module